### PR TITLE
infra: Migrate to .NET 6

### PR DESCRIPTION
**Pull request by Thog**
This migrate projects and CI to .NET 6

### CODE REVIEW

**Comment by Mou-Ikkai** [comment]
Bug: Games with high shader values crash on recompilation.
SSBU crashes at shader count (840).

**Comment by d3xMachina** [comment]
It would be interesting to see the performance difference when enabling the Dynamic PGO of .NET 6 to see if it's worth using.

**Comment by Thog** [comment]
> 
> 
> Bug: Games with high shader values crash on recompilation. SSBU crashes at shader count (840).

Should be fixed on latest commit

File: Ryujinx/Modules/Updater/Updater.cs [hunk 1]
```diff
@@ -220,7 +220,10 @@ private static void DoUpdateWithMultipleThreads(UpdateDialog updateDialog, strin
 
             for (int i = 0; i < ConnectionCount; i++)
             {
+#pragma warning disable SYSLIB0014
```
Comment (Thog) [inline Ryujinx/Modules/Updater/Updater.cs:223]: Not planning on reimplementing this logic on this PR, will create an issue once merge.

**Comment by Thog** [comment]
Ready for review.

File: Ryujinx/Modules/Updater/Updater.cs [hunk 2]
```diff
@@ -307,31 +310,56 @@ private static void DoUpdateWithMultipleThreads(UpdateDialog updateDialog, strin
...
+                    using (Stream updateFileStream = File.Open(updateFile, FileMode.Create))
+                    {
+                        long totalBytes = response.Content.Headers.ContentLength.Value;
+                        long byteWritten = 0;
+
+                        byte[] buffer = new byte[32 * 1024];
+
+                        while (true)
+                        {
+                            int writtenSize = remoteFileStream.Read(buffer);
```
Comment (Caian) [inline Ryujinx/Modules/Updater/Updater.cs:332]: Shouldn't it be readSize or sizeToWrite?

**Comment by Caian** [comment]
Games that use inline software keyboard are broken:

```
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.TypeLoadException
   at System.Drawing.SafeNativeMethods.Gdip.GdipLoadImageFromStream(IStream stream, IntPtr& image)
   at System.Drawing.Image.FromStream(Stream stream, Boolean useEmbeddedColorManagement, Boolean validateImageData)
   at Ryujinx.HLE.HOS.Applets.SoftwareKeyboard.SoftwareKeyboardRenderer.LoadResource(Assembly assembly, String resourcePath, Int32 newWidth, Int32 newHeight) in D:\a\Ryujinx\Ryujinx\Ryujinx.HLE\HOS\Applets\SoftwareKeyboard\SoftwareKeyboardRenderer.cs:line 183
   at Ryujinx.HLE.HOS.Applets.SoftwareKeyboard.SoftwareKeyboardRenderer..ctor(IHostUiTheme uiTheme) in D:\a\Ryujinx\Ryujinx\Ryujinx.HLE\HOS\Applets\SoftwareKeyboard\SoftwareKeyboardRenderer.cs:line 34
   at Ryujinx.HLE.HOS.Applets.SoftwareKeyboardApplet.Start(AppletSession normalSession, AppletSession interactiveSession) in D:\a\Ryujinx\Ryujinx\Ryujinx.HLE\HOS\Applets\SoftwareKeyboard\SoftwareKeyboardApplet.cs:line 105
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object , Span`1& , Signature , Boolean , Boolean )
   at System.Reflection.RuntimeMethodInfo.Invoke(Object , BindingFlags , Binder , Object[] , CultureInfo )
   at Ryujinx.HLE.HOS.Services.IpcService.CallHipcMethod(ServiceCtx context) in D:\a\Ryujinx\Ryujinx\Ryujinx.HLE\HOS\Services\IpcService.cs:line 122
   at Ryujinx.HLE.HOS.Services.ServerBase.Process(Int32 serverSessionHandle, UInt64 recvListAddr) in D:\a\Ryujinx\Ryujinx\Ryujinx.HLE\HOS\Services\ServerBase.cs:line 214
   at Ryujinx.HLE.HOS.Services.ServerBase.ServerLoop() in D:\a\Ryujinx\Ryujinx\Ryujinx.HLE\HOS\Services\ServerBase.cs:line 133
   at Ryujinx.HLE.HOS.Kernel.Threading.KThread.ThreadStart() in D:\a\Ryujinx\Ryujinx\Ryujinx.HLE\HOS\Kernel\Threading\KThread.cs:line 1051
```

**Comment by Mou-Ikkai** [comment]
Played SHIN MEGAMI TENSEI V for 3 hours on end. No crashes, just the generic issues already listed about the game.

**Comment by allucbr** [comment]
Tested for more than 2 hours with SMO and MK8 , no issues
it was observed a little performance improvement against Master build.


Intel i7 4770k , Linux Version RX 580.

Master build Same Place 50.4 FPS 
Master 7094

PR 54.4 FPS
Screenshot_20211114_075329

I always test all pr and master build on this place it was first time that ive get more than 53 fps, previous record was 52.

File: Ryujinx.Graphics.Gpu/Shader/Cache/CacheMigration.cs [hunk 3]
```diff
@@ -129,8 +138,11 @@ private static void RecomputeHashes(string guestBaseCacheDirectory, string hostB
                 File.WriteAllBytes(guestManifestPath, newGuestManifestContent);
                 File.WriteAllBytes(hostManifestPath, newHostManifestContent);
 
-                guestArchive.Dispose();
-                hostArchive.Dispose();
+                guestArchive.CommitUpdate();
+                guestArchive.CommitUpdate();
```
Comment (SomeoneIsWorking) [inline Ryujinx.Graphics.Gpu/Shader/Cache/CacheMigration.cs:142]: Is this right?

**Comment by Caian** [comment]
Changed the inline keyboard rendering from `System.Drawing` to `ImageSharp`

**Comment by Mou-Ikkai** [comment]
> Played SHIN MEGAMI TENSEI V for 3 hours on end. No crashes, just the generic issues already listed about the game.

8 hours now, still no issues. Actually, FPS seems to have gone up a small margin. (2-3 FPS)

**Comment by Thog** [comment]
All problems reported are now fixed

File: Ryujinx/Modules/Updater/Updater.cs (hunk 2 above)
Comment (Thog) [reply Ryujinx/Modules/Updater/Updater.cs:332]: in my mind this was the size written to the buffer but sure

**Review by emmauss — APPROVED** [review]

**Review by emmauss — APPROVED** [review]

**Review by AcK77 — APPROVED** [review]

**Pull request merged**